src/base/LoaderBase.js

```diff
@@ -0,0 +1,64 @@
+export class LoaderBase {
+
+	constructor() {
+
+		this.fetchOptions = {};
+		this.workingPath = '';
+
+	}
+
+	load( url ) {
+
+		return fetch( url, this.fetchOptions )
+			.then( res => {
+
+				if ( ! res.ok ) {
+
+					throw new Error( `Failed to load file "${ url }" with status ${ res.status } : ${ res.statusText }` );
+
+				}
+				return res.arrayBuffer();
+
+			} )
+			.then( buffer => {
+
+				if ( this.workingPath === '' ) {
+
+					this.workingPath = this.workingPathForURL( url );
+
+				}
+
+				return this.parse( buffer );
+
+			} );
+
+	}
+
+	resolveExternalURL( url ) {
+
+		if ( /^[^\\/]/.test( url ) ) {
+
+			return this.workingPath + '/' + url;
+
+		} else {
+
+			return url;
+
+		}
+
+	}
+
+	workingPathForURL( url ) {
+
+		const splits = url.split( '/' );
```

Comment:
Leaving a note here so I don't forget -- it looks like you fixed it in a different way but from #229 it seems that the working path was not being set from the loaded url correctly because it was splitting on `\/` rather that `/` or `\`. But can we change this split to use the `/[\\/]/g` regex so it works with both forward and backward slashes? I'd do it in another PR but I'd like to avoid conflicts!

And thanks again for the work! I'll be able to take a more detailed look early next week.